storage/innobase/include/ut0ut.h

```diff
@@ -88,15 +88,26 @@ struct ut_when_dtor {
    the YieldProcessor macro defined in WinNT.h. It is a CPU architecture-
    independent way by using YieldProcessor. */
 #  define UT_RELAX_CPU() YieldProcessor()
-# elif defined(HAVE_ATOMIC_BUILTINS)
+# elif defined(__powerpc__)
+#include <sys/platform/ppc.h>
 #  define UT_RELAX_CPU() do { \
-     volatile lint	volatile_var; \
-     os_compare_and_swap_lint(&volatile_var, 0, 1); \
+     volatile lint      volatile_var = __ppc_get_timebase(); \
    } while (0)
 # else
 #  define UT_RELAX_CPU() ((void)0) /* avoid warning for an empty statement */
 # endif
 
+#define UT_COMPILER_BARRIER() __asm__ __volatile__ ("":::"memory")
```

Comment:
Looks gcc-ish   The correspording intrinsic for Microsoft compiler is called MemoryBarrier()